pkg/login/social/google_oauth.go

```diff
@@ -48,3 +116,95 @@ func (s *SocialGoogle) AuthCodeURL(state string, opts ...oauth2.AuthCodeOption)
 	}
 	return s.SocialBase.AuthCodeURL(state, opts...)
 }
+
+func (s *SocialGoogle) extractFromToken(ctx context.Context, client *http.Client, token *oauth2.Token) (*googleUserData, error) {
+	s.log.Debug("Extracting user info from OAuth token")
+
+	idToken := token.Extra("id_token")
+	if idToken == nil {
+		s.log.Debug("No id_token found, defaulting to API access", "token", token)
+		return nil, nil
+	}
+
+	rawJSON, err := s.retrieveRawIDToken(idToken)
+	if err != nil {
+		s.log.Warn("Error retrieving id_token", "error", err, "token", fmt.Sprintf("%+v", idToken))
+		return nil, nil
+	}
+
+	s.log.Debug("Received id_token", "raw_json", string(rawJSON))
+
+	var data googleUserData
+	if err := json.Unmarshal(rawJSON, &data); err != nil {
```

Comment:
question: is it possible that the ID token is in the legacy format?

Reply:
seems not, it's a openID standard compliant https://openid.net/specs/openid-connect-core-1_0.html#IDToken